### PR TITLE
Add test that Gymnasium and MO-Gymnasium envs match

### PR DESCRIPTION
**Pull request by pseudo-rnd-thoughts**
This PR adds a test for all MO-Gymnasium envs that are also contained in Gymnasium to see if the two are still equivalent using the `check_environments_match` function

### CODE REVIEW

File: tests/test_envs.py [hunk 1]
```diff
@@ -40,6 +41,32 @@ def test_all_env_passive_env_checker(spec):
...
+        ("HalfCheetah-v4", "mo-halfcheetah-v4"),
+        ("Walker2d-v4", "mo-walker2d-v4"),
+        ("Ant-v4", "mo-ant-v4"),
+        ("Swimmer-v4", "mo-swimmer-v4"),
+        ("Humanoid-v4", "mo-humanoid-v4"),
+    ],
+)
+def test_gymnasium_equivalence(gym_id, mo_gym_id, num_steps=100, seed=123):
+    env = gym.make(gym_id)
+    mo_env = mo_gym.LinearReward(mo_gym.make(mo_gym_id))
```
Comment (ffelten) [inline tests/test_envs.py:61]: I have to check but I don't think all our environments just "split" the rewards into components

**Comment by pseudo-rnd-thoughts** [comment]
@ffelten or @LucasAlegre It looks like most of the mo-gymnasium reward vectors do not by default match the gymnasium versions. Is this expected?

**Comment by ffelten** [comment]
> @ffelten or @LucasAlegre It looks like most of the mo-gymnasium reward vectors do not by default match the gymnasium versions. Is this expected?

Lol, just what I said in my comment. Some environments are indeed different, for example, we usually remove the scaling factors because they are unnecessary since the rewards are not aggregated into a single scalar. Another example is environments that add an extra component to the reward, e.g. mountaincar penalizes for changing directions.

**Comment by pseudo-rnd-thoughts** [comment]
> > @ffelten or @LucasAlegre It looks like most of the mo-gymnasium reward vectors do not by default match the gymnasium versions. Is this expected?
> 
> Lol, just what I said in my comment. Some environments are indeed different, for example, we usually remove the scaling factors because they are unnecessary since the rewards are not aggregated into a single scalar. Another example is environments that add an extra component to the reward, e.g. mountaincar penalizes for changing directions.
> 
> EDIT: still, I think these kinds of tests are relevant, I'd just ignore the reward tests.

Amazing, I'm glad this isn't actually an issue then. 
Is there a way of knowing the reward weightings to make it equivalent to the Gymnasium version?

**Comment by LucasAlegre** [comment]
> > > @ffelten or @LucasAlegre It looks like most of the mo-gymnasium reward vectors do not by default match the gymnasium versions. Is this expected?
> > 
> > 
> > Lol, just what I said in my comment. Some environments are indeed different, for example, we usually remove the scaling factors because they are unnecessary since the rewards are not aggregated into a single scalar. Another example is environments that add an extra component to the reward, e.g. mountaincar penalizes for changing directions.
> > EDIT: still, I think these kinds of tests are relevant, I'd just ignore the reward tests.
> 
> Amazing, I'm glad this isn't actually an issue then. Is there a way of knowing the reward weightings to make it equivalent to the Gymnasium version?

Yes, I believe for most of them (if not all) that is indeed possible. I will try to document this somewhere, any suggestion?

**Comment by pseudo-rnd-thoughts** [comment]
In the documentation? 

Are you aware that `Reacher` action space is different from the Gymnasium one. Otherwise, I

**Comment by LucasAlegre** [comment]
> In the documentation?

I will add to the pydoc of each environment class.

> Are you aware that `Reacher` action space is different from the Gymnasium one?

Yes, Reacher is probably the environment with the most changes from the original.  We also changed .xml to add more targets to the environment. This one is not possible to map to Reacher from Gymnasium

**Comment by pseudo-rnd-thoughts** [comment]
> I will add to the pydoc of each environment class.

Thanks, this will be really helpful for me because I want to learn the equivalent reward vector version of the standard agent which requires that the mo version is equivalent to the standard version

**Comment by LucasAlegre** [comment]
@pseudo-rnd-thoughts see:
https://github.com/Farama-Foundation/MO-Gymnasium/pull/92

I could not map the rewards for Humanoid and Walker2d because we added the healthy_reward to all objectives, and then we can not scale it separately. What we could do is create versions of them with the healthy_reward modeled as a separate objective. What do you think? @ffelten

**Comment by ffelten** [comment]
I find it odd to create environments for the sake of testing lol

**Comment by LucasAlegre** [comment]
> I find it odd to create environments for the sake of testing lol

It is not only for testing, since I believe the relative weighting of the healthy_reward might also induce new trade-offs. I saw some papers using it. The reason I did not include it was to focus on the velocity/energy trade-off, which is clearer.

**Comment by LucasAlegre** [comment]
I figure it out how to scale the healthy_reward so that Humanoid and Walker2d can be mapped to the original Gymnasium env. But this will make the current results not reproducible, so I will implement these changes on v5 (PR #85).

**Comment by pseudo-rnd-thoughts** [comment]
@ffelten or @LucasAlegre can we merge this PR and the reward vector equivalent be in another PR?

**Review by ffelten — APPROVED** [review]

**Pull request merged**